### PR TITLE
Add Pytest as test suite, to Jumpy.

### PR DESCRIPTION
**Pull request by akshaysharma096**
These are the changes made.

- Add Pytest as the test suite.
- Some code was still not in PEP8 format, eg: `memory_manager.py` and `array_creation.py`, I have renamed `array_creation.py` to `test_array_creation.py` and ported  `memory_manager.py` to PEP8.
- Updated setup.py
- Updated requirements.py
- Add license as doc string in all code.

### CODE REVIEW

File: jumpy/requirements.txt [hunk 1]
```diff
@@ -1,4 +1,9 @@
+attrs==18.1.0
 Cython==0.28.2
+more-itertools==4.1.0
```
Comment (maxpumperla) [inline jumpy/requirements.txt]: Excuse my ignorance, but I don't see (from your PR) why we need most of these new requirements. @akshaysharma096

Comment (akshaysharma096) [reply jumpy/requirements.txt]: @maxpumperla ,  `more-itertools` is in the dependency of pytest testing suite, we can verify from https://github.com/pytest-dev/pytest/blob/master/setup.py.

Thanks!

**Comment by akshaysharma096** [comment]
@AlexDBlack @maxpumperla , can we merge ?

File: jumpy/requirements.txt (hunk 1 above)
Comment (maxpumperla) [reply jumpy/requirements.txt]: exactly, but you get them with pytest. no need for attr etc.

**Comment by maxpumperla** [comment]
remove redundant dependencies, then we can merge!

**Comment by akshaysharma096** [comment]
@maxpumperla sure, will do.

**Comment by AlexDBlack** [comment]
Thanks :)

**Pull request merged**